src/modules/bootloader/main.py

```diff
@@ -637,28 +637,33 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--bootloader-id=" + efi_bootloader_id,
                                    "--force"])
     else:
-        assert efi_directory is None
-        if libcalamares.globalstorage.value("bootLoader") is None:
-            return
+        if efi_directory is not None:
+            if not install_hybrid_grub:
+                return
 
-        boot_loader = libcalamares.globalstorage.value("bootLoader")
-        if boot_loader["installPath"] is None:
-            return
+        if libcalamares.globalstorage.value("bootLoader") is None:
+            find_esp_disk_command = "echo \"/dev/$(lsblk -o PKNAME $(df --output=source /boot/efi | tail -n1) | tail -n1)\""
```

Comment:
We shouldn't hardcode `/boot/efi`

The efi mountpoint is configurable.

Reply:
gah, I knew that, but forgot it here. Will fix in the next push.

Reply:
Um, wait, the efi mountpoint *is* configurable, but I'm pretty sure that's stored in `libcalamares.globalstorage.value("bootLoader")["installPath"]` no? This is the codepath for if `libcalamares.globalstorage.value("bootLoader") is None`, in which case we don't have that info. Is there another convenient spot to grab it from, or should I be parsing the partition layout to find the right spot and falling back to `/boot/efi` if all else fails?

Reply:
I think it is also in global storage.  Something like `efiSystemPartition`

Reply:
Was able to use `efiSystemPartition`. `/boot/efi` is still used as a final fallback, but only if all other options fail.